meinberlin/apps/contrib/assets/CountDown.tsx

```diff
@@ -1,6 +1,13 @@
 import React from 'react'
 
-export const CountDown = (props) => {
+interface CountDownProps {
+  activeClass: string
+  inactiveClass: string
+  countText: string
+  counter: number
+}
+
+export const CountDown: React.FC<CountDownProps> = (props) => {
```

Comment:
for me this is already not straightforward to understand tbh. After staring at it a while i get the idea of this typing, but not sure if this will be a pain in the bigger scale?

I would expect something like this:

```
export const CountDown: React.FC = (props: CountDownProps) => {
    ...
}
```

even better if we could omit the component type to have less code pollution, imho it doesn't give
much information to the dev:

```
export const CountDown = (props: CountDownProps) => {
    ...
}
```
Can it be inferred like this?

Reply:
what do you think @goapunk and what do the others think @phillimorland @fuzzylogic2000 @Rineee ?

Reply:
@khamui I agree that the return type is annoying, I initially had the second option you posted but the linter was complaining about the missing return type. From what I understand typescript will infer `JSX.Element` instead of `React.FC`, so I guess the question is whether we want the extra type checking we get by adding `React.FC` or we prefer simpler syntax. Option 1 doesn't work for me.

I'm torn between the easier syntax and the proper type checking, but currently tend to say we should probably go for the strict-but-ugly way

Reply:
I do have a preference for easier syntax although looking through some typescript examples, if we decide to go with it, I feel it will not be particularly readable.

Reply:
@phillimorland i agree that we have to get use to some of typescript words and "syntax", but i am really torn about additions that does not provide any further information. In this particular case `React.FC` does not help us. We know by the function component notation that it is a `FC`. In those cases i would prefer a config, which allows us to omit this unnecessary additions. I am afraid that it makes our code harder to read, to understand and that it slows us down in the end.

@goapunk regarding the React.FC i found this from Facebook/React:
https://github.com/facebook/create-react-app/pull/8177

i find the following better, would that work?
`const Component1 = (props: CountDownProps): JSX.Element => { /*...*/ }`

similar from here https://stackoverflow.com/a/61547010

Reply:
The biggest point people critize about `React.FC` seems to be as in your link: `Provides an implicit definition of children` which is no longer true for react 18. I do agree that we can go without it however, question then is if we want to add the JSX.Element or infer it